[receiver/azureblob] Support service principal authentication for Azure Blob storage (#32706)
open-telemetry/opentelemetry-collector-contrib: receiver/azureblobreceiver/README.md

The following settings are required:

- `event_hub:`
  `  endpoint:` (no default): Azure Event Hub endpoint triggering on the `Blob Create` event 

The following settings can be optionally configured:

- `auth` (default = connection_string): Specifies the used authentication method. Supported values are `connection_string`, `service_principal`.

> - `cloud` (default = "AzureCloud"): defines which Azure cloud to use when using the `service_principal` authentication method. Either `AzureCloud` or `AzureUSGovernment`

Reviewer: ```suggestion
- `cloud` (default = "AzureCloud"): Defines which Azure Cloud to use when using the `service_principal` authentication method. Value is either `AzureCloud` or `AzureUSGovernment`.
```

- `logs:`
  `  container_name:` (default = "logs"): Name of the blob container with the logs
- `traces:`
  `  container_name:` (default = "traces"): Name of the blob container with the traces

> Authenticating using a connection string requires following additional settings:

Reviewer: ```suggestion
Authenticating using a connection string requires configuration of the following additional setting:
```

- `connection_string:` Azure Blob Storage connection key, which can be found in the Azure Blob Storage resource on the Azure Portal.

> Authenticating using service principal requires following additional settings:

Reviewer: ```suggestion
Authenticating using service principal requires configuration of the following additional settings:
```

- `tenant_id`
- `client_id`
- `client_secret`
- `storage_account_url`

> The service principal requires the [Storage Blob Data Contributor](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/storage#storage-blob-data-contributor) role on the logs and traces containers.

Reviewer: ```suggestion
The service principal method also requires the [Storage Blob Data Contributor](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/storage#storage-blob-data-contributor) role on the logs and traces containers.
```

> ### Example Configurations

Reviewer: ```suggestion
### Example configurations
```

Using connection string for authentication:

```yaml
receivers:
  azureblob:
    connection_string: DefaultEndpointsProtocol=https;AccountName=accountName;AccountKey=+idLkHYcL0MUWIKYHm2j4Q==;EndpointSuffix=core.windows.net
    event_hub:
      endpoint: Endpoint=sb://oteldata.servicebus.windows.net/;SharedAccessKeyName=otelhubbpollicy;SharedAccessKey=mPJVubIK5dJ6mLfZo1ucsdkLysLSQ6N7kddvsIcmoEs=;EntityPath=otellhub
```

> Using Service Principal for authentication:

Reviewer: ```suggestion
Using service principal for authentication:
```
